ADR/0022-secret-mgmt-for-user-workloads.md

```diff
@@ -380,6 +381,27 @@ status:
     status: "False"
     type: DataObtained
 ```
+### Internal RemoteSecrets
+The RemoteSecret Custom Resource (CR) is a crucial implementation detail of "Secret Management For User Workloads". It is also utilized in other features within RHTAP, such as "pull" secrets, among others.
+To differentiate RemoteSecrets that are provided by end-users from RemoteSecrets that are generated by RHTAP's components, a label `appstudio.redhat.com/internal: true` is introduced. The presence of this label signifies that the RemoteSecret is intended for internal usage, and any unauthorized changes to such secrets may result in unpredictable outcomes.
+This label helps to identify secrets that are meant for internal purposes, preventing accidental or unauthorized modifications that could disrupt the functioning of RHTAP's components.
+
+Example:
+
+```yaml
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
+metadata:
+    name: test-remote-secret-secret
+    labels:
+        appstudio.redhat.com/component: m-service
+        appstudio.redhat.com/application: coffee-shop
+        appstudio.redhat.com/internal: true
+spec:
+...
+```
+It's important to note that this label is not intended to alter the behavior of the controller or permissions. The primary effect is on the UI behavior, which is designed to hide these labeled RemoteSecrets from the list.
```

Comment:
This will not protect workspsace Admins from deleting these internal RemoteSecrets from the CLI as they would have the ability to be destructive with `get, list, watch, create, update, patch, delete` permissions, correct?

Reply:
@mmorhun, are you aware of these plans?

Reply:
Partially. There was a discussion with Gal, but no action items requested so far. There are pros and cons that should be discussed. Anyway, this is an off topic for this ADR.

Reply:
@mmorhun @skabashnyuk , would one of you be able to create an issue for that work/investigation with necessary details so that we don't lose track of the request?

Reply:
We can create a Spike for the discussion and stories after that as/if needed.